### ISSUE TITLE
bhilare_ - While creating an Auction, routing information in `lotRouting` is never stored for new `lotId`, instead is overwritten for lotId = 0.

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-03-axis-finance-judging#47]
bhilare_

high

# While creating an Auction, routing information in `lotRouting` is never stored for new `lotId`, instead is overwritten for lotId = 0.

## Summary
While creating an Auction , `Routing` information for the new lot, which are stored in `lotRouting` mapping, is not stored for the new lotId, instead it is overwritten for lotId=0.

## Vulnerability Detail
A new auction in being created from `Auctioneer::auction` function as you can see below 

https://github.com/sherlock-audit/2024-03-axis-finance/blob/cadf331f12b485bac184111cdc9ba1344d9fbf01/moonraker/src/bases/Auctioneer.sol#L160-L285

Now as you can see in L-164 , the lotId parameter is being declared in the returns part.

```javascript
 ) external nonReentrant returns (uint96 lotId) {
```
That means the value initially will be 0.
And then at line L-174 :
```javascript
Routing storage routing = lotRouting[lotId];
```
The routing information is loaded to store using the lotId, that is currently 0.
That means everytime an auction will be created thing `routing` storage variable will be loading the routing information of lotId = 0 , always and the auction functions would be making storage levels changes for the lotId = 0 's routing information only.
## Impact
The routing information for the current lotId is never stored and only lotId= 0 's routing information is always reverted.

Impact is very severe because all future operations where the routing information will be needed, will not work as intended, as there's no information stored for the current lotId.

For example if you want to `cancel` auction, you can see at L-305 in `Auctioneer` contract

```javascript
Routing storage routing = lotRouting[lotId_];
```
The routing information is being loaded for the provided `lotId_`, but since it was never stored while creating the auction , there would no correct information, and hence further steps would revert. Hence would never be able to cancel.

Wherever the storage mapping variable `lotRouting` would be stored, will work in wrong way, whether be it while cancelling auction or getting the module for the lot using `_getModuleForId` function.

```javascript
function _getModuleForId(uint96 lotId_) internal view returns (AuctionModule) {
        // Load module, will revert if not installed
        return AuctionModule(_getModuleIfInstalled(lotRouting[lotId_].auctionReference));
    }
```
## Code Snippet
https://github.com/sherlock-audit/2024-03-axis-finance/blob/cadf331f12b485bac184111cdc9ba1344d9fbf01/moonraker/src/bases/Auctioneer.sol#L174
## Tool used
Manual Review

## Recommendation
Incrementing the lot count and getting id is done at line L-194
```javascript
            // Increment lot count and get ID
            lotId = lotCounter++;
```

This step should be done before loading the routing information for lotId using `lotRouting` mapping, & then routing information should be loaded for making storage level changes and storing the information.

Duplicate of #12
